packages/amplify_datastore_plugin_interface/lib/src/types/models/model_field_definition.dart

```diff
@@ -101,18 +100,18 @@ class ModelFieldDefinition {
   }
 
   static ModelFieldDefinition belongsTo(
-      {@required QueryField key,
+      {required QueryField key,
       bool isRequired: true,
-      @required String ofModelName,
-      QueryField associatedKey,
-      String targetName}) {
+      required String ofModelName,
+      QueryField? associatedKey,
+      String? targetName}) {
     // Extra code needed due to lack of nullability support
-    String associatedName;
-    String associatedType;
+    String? associatedName;
+    String? associatedType;
 
     associatedName = associatedKey?.fieldName;
     associatedType = associatedKey?.fieldType != null
-        ? associatedKey.fieldType.ofModelName
+        ? associatedKey!.fieldType!.ofModelName
         : ofModelName;
```

Comment:
Not sure if we can do the following, but would be cool if we can.

```dart
associatedType = associatedKey?.fieldType?.ofModelName ?? ofModelName;
```

Reply:
that works!  added that in